extension/src/handlers/video/record-media-handler.ts

```diff
@@ -112,10 +112,12 @@ export default class RecordMediaHandler {
 
             try {
                 const audioBase64 = await audioPromise;
-                audioModel = {
-                    ...baseAudioModel,
-                    base64: audioBase64,
-                };
+                if (audioBase64 !== '') {
```

Comment:
If we allow this, the `audioModel` can be  silently `undefined`. In what situation is the return value turning up empty?

Edit: I read farther below and I see that you are returning empty string intentionally. I think an exception should be thrown instead.